File: src/main/resources/crawling/device.py
from sqlalchemy import Column, Integer, String , Text
from sqlalchemy.ext.declarative import declarative_base

Base = declarative_base()

class Device(Base):
    __tablename__ = 'device'

    id = Column(Integer, primary_key=True, autoincrement=True)
    device_name =Column(String(255))#제품명
    color = Column(String(255))#색상
    storage_capacity = Column(String(255))#용량 메모리 , 용량 따로 빼기 
    width = Column(String(50))#가로
    length = Column(String(50))#세로
    height = Column(String(50))#높이
    weight = Column(String(50))#무게
    display = Column(String(255))# 디스플레이 
    diagonal = Column(String(50))# 대각선 길이
    resolution = Column(String(50))# 해상도
    chip = Column(String(100))# 프로세서 
    release_date = Column(String(255))#설명
    images = Column(Text)
    #연결 단자 

File: src/main/java/Matdol/SmartGazalBee/DeviceComparison/Domain/Device.java
package Matdol.SmartGazalBee.DeviceComparison.Domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Device {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "device_name") //,unique = true
    private String deviceName; // 제품명
    private String color; //색상
    @Column(name = "storage_capacity")
    private String storageCapacity; // 용량 메모리, 용량 따로 빼기
    private String weight; // 무게
    private String width; // 가로
    private String length; // 세로
    private String height; // 두깨, 높이
    private String display; // 디스플레이 크기
    private String diagnoal; // 대각선
    private String resolution; // 해상도
    private String chip; // 프로세서
    @Column(name = "release_date")
    private String releaseDate; // 출시일
    private String images;

}
